backend/utils/feed.utils.js

```diff
@@ -0,0 +1,158 @@
+    await Promise.all(
+      user.ActivityLogs.map(async (activity) => {
+        if (activity.relatable_type === "quizzes") {
+          await Quiz.findByPk(activity.relatable_id, { paranoid: false }).then(
+            (quiz) => {
+              if (currentUserId === targetUserId) {
+                result = [
+                  ...result,
+                  {
+                    activity: `You learnt ${quiz.name}`,
+                    avatar_url: user.avatar_url,
+                    timestamp: activity.updatedAt,
+                  },
+                ];
+              } else {
+                result = [
+                  ...result,
+                  {
+                    activity: `${user.first_name} learnt ${quiz.name}`,
+                    avatar_url: user.avatar_url,
+                    timestamp: activity.updatedAt,
+                  },
+                ];
+              }
+            }
+          );
+        } else if (activity.relatable_type === "follows") {
+          const id = await Follow.findByPk(activity.relatable_id, {
+            attributes: ["following_id"],
+          });
+          await getUser(id.following_id).then(async (uInfo) => {
+            if (currentUserId === targetUserId) {
+              result = [
+                ...result,
+                {
+                  activity: `You followed ${uInfo.first_name}`,
+                  avatar_url: user.avatar_url,
+                  timestamp: activity.updatedAt,
+                },
+              ];
+            } else {
+              result = [
+                ...result,
+                {
+                  activity: `${user.first_name} followed ${uInfo.first_name}`,
+                  avatar_url: user.avatar_url,
+                  timestamp: activity.updatedAt,
+                },
+              ];
+            }
+          });
+        } else if (activity.relatable_type === "unfollows") {
+          const id = await Follow.findByPk(activity.relatable_id, {
+            attributes: ["following_id"],
+          });
+          await getUser(id.following_id).then(async (uInfo) => {
+            if (currentUserId === targetUserId) {
+              result = [
+                ...result,
+                {
+                  activity: `You unfollowed ${uInfo.first_name}`,
+                  avatar_url: user.avatar_url,
+                  timestamp: activity.updatedAt,
+                },
+              ];
+            } else {
+              result = [
+                ...result,
+                {
+                  activity: `${user.first_name} unfollowed ${uInfo.first_name}`,
+                  avatar_url: user.avatar_url,
+                  timestamp: activity.updatedAt,
+                },
+              ];
+            }
+          });
+        }
+      })
+    );
+    return result;
+  } catch (err) {
+    throw new Error({ error: "Something went wrong, please try again later" });
+  }
+};
+
+const getLearntWordsAndLessons = async (targetUserId) => {
```

Comment:
- move this in model since this is a model related. Create a function inside a model and move these code.
sample implementation:
```
const { Model } = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class User extends Model {
    static associate(models) {
      ...
    }
  }
  User.init(
    ...
  );

  User.getLearntWordsAndLessons = function () {
    console.log("here are the learnt words and lessons");
  };

  return User;
};

```